py3status/modules/xsel.py

```diff
@@ -32,13 +38,25 @@ class Py3status:
     """
     # available configuration parameters
     cache_timeout = 0.5
-    command = 'xsel -o'
+    command = 'xsel -output'
+    click_command = 'xsel --clear --clipboard'
     format = '{selection}'
     max_size = 15
     symmetric = True
+    log_file = None
+
+    selection_cache = None
 
     def xsel(self):
-        selection = self.py3.command_output(self.command)
+        selection = self.py3.command_output(self.command).strip()
+
+        if self.log_file and selection and selection != self.selection_cache:
```

Comment:
(Optional) For every `i3-msg restart`, the log will get a date with `(selection)`. Sometimes same `(selection)`. We could be a bit clever by reading the last `(selection)` in the file instead of using `self.selection_cache = None` in the `post_config_hook`.

Reply:
But I would still keep `self.selection_cache = None` to avoid file reads all the time. I would just add another case to the if at the end to check whether the current `selection` is already present. This should only read the file when py3status starts up.

Reply:
If I restart `i3` occasionally to play with `py3status` modules, my log would looks like this.
```
2017-10-23 06:48:00
~
2017-10-23 06:48:18
~
2017-10-23 06:48:27
~
2017-10-23 06:48:36
~
2017-10-23 06:48:45
~
2017-10-23 06:48:59
~
```